### PR TITLE
feat!: Use same type for flag metadata and event metadata

### PR DESCRIPTION
**Pull request by benjiro**
## This PR

**BREAKING CHANGE**, merge with 2.0

The spec describes two types(flag metadata, and event metadata) that are functionally the same. This PR makes a breaking change to bring both of the types to use a generic ImmutableMetadata type.

- Rename BaseMetadata to ImmutableMetadata, make sealed class and implement a empty constructor

### Related Issues

Fixes: #234

### CODE REVIEW

File: src/OpenFeature/Model/ImmutableMetadata.cs [hunk 1]
```diff
@@ -5,13 +5,27 @@
 namespace OpenFeature.Model;
 
 /// <summary>
-/// Represents the base class for metadata objects.
+/// Represents the a immutable metadata associated with a feature flags and events.
```
Comment (kinyoklion) [inline src/OpenFeature/Model/ImmutableMetadata.cs:8]:
```suggestion
/// Represents immutable metadata associated with feature flags and events.
```

**Comment by beeme1mr** [comment]
Hey @benjiro, we started prepping for the 2.0 release. When you have a moment, could you please resolve the conflicts? Thanks!

**Comment by toddbaert** [comment]
I've taken this up where @benjiro left off, fixed some conflicts, etc.

**Review by askpt — APPROVED** [review]
Looks good to me. The delta changes looks bigger but it is only formatting related. 👍

**Review by kinyoklion — APPROVED** [review]

**Review by toddbaert — APPROVED** [review]

**Pull request merged**